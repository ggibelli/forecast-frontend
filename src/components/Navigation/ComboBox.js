import React from 'react'
import TextField from '@material-ui/core/TextField'
import Autocomplete from '@material-ui/lab/Autocomplete'
import { useSelector } from 'react-redux'

export default function Grouped() {
  const surfSpots = useSelector((state) => state.spotsSearch)
  const options = surfSpots.map((spot) => {
    const region = spot.region.name
    const { name } = spot
    return {
      region,
      name,
    }

  })
  return (
    <Autocomplete
      id="grouped"
      options={options.sort((a, b) => a.region.localeCompare(b.region))}
      
      getOptionLabel={(spot) => spot.name}
      groupBy={(spot) => spot.region}
      style={{ width: 300 }}
      renderInput={(params) => (
        <TextField {...params} label="surf spot.." variant="filled" />
      )}
    />
  )
}
