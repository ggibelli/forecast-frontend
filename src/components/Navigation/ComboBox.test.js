import React from 'react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import renderer from 'react-test-renderer'

import ComboBox from './ComboBox'

const mockStore = configureMockStore([thunk])
let store
let component

describe('ComboBox', () => {
  beforeEach(() => {
    store = mockStore({
      spotsSearch: [
        {
          name: 'abc',
          continent: { name: 'cbd', id: '234' },
          country: { name: 'adf', id: '445' },
          region: { name: 'lol', id: '090' },
          id: '123',
        },
        {
          name: 'qwe',
          continent: { name: 'rty', id: '123' },
          country: { name: 'asd', id: '678' },
          region: { name: 'poi', id: '987' },
          id: '235',
        },
      ],
    })
    component = renderer.create(
      <Provider store={store}>
        <ComboBox />
      </Provider>,
    )
  })
  test('renders ComboBox component', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })
})
