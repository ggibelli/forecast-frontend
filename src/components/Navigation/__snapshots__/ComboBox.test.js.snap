// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComboBox renders ComboBox component 1`] = `
<div
  aria-expanded={false}
  aria-owns={null}
  className="MuiAutocomplete-root makeStyles-search-1 MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  role="combobox"
>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
      data-shrink={false}
      htmlFor="grouped"
      id="grouped-label"
    >
      surf spot..
    </label>
    <div
      className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiFilledInput-adornedEnd"
      onClick={[Function]}
    >
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-controls={null}
        aria-invalid={false}
        autoCapitalize="none"
        autoComplete="off"
        autoFocus={false}
        className="MuiInputBase-input MuiFilledInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiFilledInput-inputAdornedEnd"
        disabled={false}
        id="grouped"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        required={false}
        spellCheck="false"
        type="text"
        value=""
      />
      <div
        className="MuiAutocomplete-endAdornment"
      >
        <button
          aria-label="Clear"
          className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          title="Clear"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
        </button>
        <button
          aria-label="Open"
          className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          title="Open"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
